### ISSUE TITLE
Smaller base learning rate for bias and scale

### ISSUE THREAD

**Issue by kumasento** [hongyi-zhang/Fixup#6]
Hi, thanks for releasing this high-quality codebase! It is really helpful to understand your excellent paper.

I am trying to re-implement it for other purposes, and I just noticed that you assign a 10x smaller initial learning rate for bias and multiplier terms. It seems to be necessary, since in my implementation, setting their learning rate to `0.1` will cause divergence somewhere in the middle of the training.

I'm just wondering whether this is the reason that you decided to scale down the learning rate for bias and multiplier terms. And if not, is there any other insight?

Thank you in advance for any clarification.

Best regards
Ruizhe

**Comment by hongyi-zhang** [comment]
Hi Ruizhe,

Yes, scaling down the learning rate of scalars is for improving the stability of training. Three reasons support using a smaller learning rate for the scalar parameters:

1. The gradient of loss function w.r.t. a scalar parameter aggregates gradient components across all channels of the convolutional layer, making it relatively more prone to gradient explosion than the conv weights.

2. Correspondingly, the update to a scalar parameter affects the activations of all channels in the next convolutional layer, thus having a relatively greater impact on the network output than the conv weights.

3. There are only O(L) scalar parameters, where L is the network depth and typically much smaller than the size of the training set. Hence there is very little risk of overfitting due to using a small learning rate on the scalar parameters.

Hope this helps to clarify!

**Comment by kumasento** [comment]
Hi Hongyi,

Thanks for your detailed reply! I'm all cleared now :)
